lib/dal/src/job/definition/action.rs

```diff
@@ -165,7 +166,20 @@ async fn inner_run(
             }
         }
     }
+
     ctx.commit().await?;
+
+    // Now that everything has passed, we can send billing events.
+    match prototype.kind {
```

Comment:
Nice, this should cut down on the number of unnecessary events we send!

Reply:
Yep!